ocean_lib/web3_internal/event_listener.py

```diff
@@ -52,12 +56,12 @@ def make_event_filter(self):
 
     def listen_once(
         self,
-        callback,
-        timeout=None,
-        timeout_callback=None,
-        start_time=None,
-        blocking=False,
-    ):
+        callback: Optional[Callable] = None,
```

Comment:
Did you intentionally make the `callback` optional?  The underlying `watch_one_event` Thread seems to require a non-None `Callable`.

Reply:
yes, but the event can be non-blocking. And this was the input detected from the tests. I assume it's ok like this but I will revisit this.

Reply:
I don't have a definitive answer on this. I believe it should be kept Optional, but I am open to arguments.

Reply:
Taking a second look at this, I agree: `Optional` is a more appropriate design. 

But the underlying `watch_one_event` call requires a non-None `Callable`. I now believe that the correct resolution is to add a `None` check before calling `callable()` and change the `callable` parameter to `Optional[Callable]`

Reply:
I think that check is done implicitly. There is this instruction `original_callback = callback` and then a check `if original_callback:` It also looks like this function is a bit half-baked, with the `#TODO`, so I wouldn't change a lot about it without knowing more. Maybe it warrants a separate issue? Not sure.

Reply:
Ah, you're right. Sorry I missed this. `listen_once` defines the nested function `_callback` which is passed into `watch_one_event`, thus ensuring the `callback` parameter is never `None`. I think a separate issue is not warranted.